src/vs/workbench/browser/parts/editor/textDiffEditor.ts

```diff
@@ -132,7 +133,8 @@ export class TextDiffEditor extends BaseTextEditor implements ITextDiffEditor {
 			this.diffNavigatorDisposables.push(this.diffNavigator);
 
 			// Readonly flag
-			diffEditor.updateOptions({ readOnly: resolvedDiffEditorModel.isReadonly() });
+			// {{SQL CARBON EDIT}}
+			diffEditor.updateOptions({ readOnly: isReadOnly ? true : resolvedDiffEditorModel.isReadonly() });
```

Comment:
the `resolvedDiffEditorModel.isReadonly()` comes from the input, therefore you should be able to control your input and not need to make changes here, vscode has prior art of diffs being readonly so there should be a way.

Reply:
UntitledEditorInput.isReadOnly() is hardcoded to return false, so I switched to use ResourceEditorInput since it is readonly.